File: app/javascript/packs/tag.js
import Taggle from 'taggle'

$(document).ready(() => {
  if ($('#article-tag-form').length) {
    new Taggle('article-tag-form', {
      tags: [],
      hiddenInputName: 'article[tags_attributes][name][]'
    })
  }
});

File: app/models/tag.rb
# == Schema Information
#
# Table name: tags
#
#  id         :bigint           not null, primary key
#  name       :string(255)      not null
#  created_at :datetime         not null
#  updated_at :datetime         not null
#
class Tag < ApplicationRecord
  has_many :article_tag, dependent: :destroy
  has_many :article, through: :article_tag

  validates :name, presence: true
end
